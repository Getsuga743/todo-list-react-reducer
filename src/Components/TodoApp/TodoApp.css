.done{
    color: grey;
    text-decoration:line-through;
}
.todo{
    color:hsl(0, 0%, 98%);
    margin-top:4rem;
    word-spacing: 1rem;
    font-size: 1.5rem;
}
.todo-title{
    letter-spacing: 1rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
}
.todo-title span img{
    width: 100%;
}
.todo-title span {
    width:3rem;
}

@media only screen and (max-width: 500px) {
 .todo{
     margin-top: 2rem;
     font-size: 1rem;
 }
 .todo-title{
    letter-spacing: 1rem;
    display: flex;
    flex-direction: row;
    align-items: flex-end;
}
.todo-title span{
    margin:auto;
}
.todo-title span:first-child{
    width:100%;
}
.todo-title span:first-child h1{
    display:inline-flex;
    align-items:flex-end;
    flex-direction: column;
    justify-content:flex-end;
    margin:0;
}
.todo-title span:last-child{
    display: inline-flex;
    width: 2rem;
}
}