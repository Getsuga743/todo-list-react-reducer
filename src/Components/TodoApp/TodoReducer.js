/* eslint-disable no-return-assign */
/* eslint-disable no-param-reassign */
const todoReducer = (state = [], action) => {
  switch (action.type) {
    case 'add':
      return state = [...state, action.payload];
    default:
      return state;
  }
};
export default todoReducer;
