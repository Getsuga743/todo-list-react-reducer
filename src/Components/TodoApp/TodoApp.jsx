/* eslint-disable no-undef */
/* eslint-disable no-unused-vars */
/* eslint-disable no-unused-expressions */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { useReducer, useEffect } from 'react';
import todoReducer from './TodoReducer';
import TodoList from '../Todo-list/TodoList';
import './TodoApp.css';
import AddTodo from '../AddTodo/AddTodo';
import img from '../../images/icon-sun.svg';

const init = () => JSON.parse(localStorage.getItem('todos')) || [];

const TodoApp = () => {
  const [todos, dispatch] = useReducer(todoReducer, [], init);

  useEffect(() => {
    localStorage.setItem('todos', JSON.stringify(todos));
    // return () => {
    //   cleanup
    // }
  }, [todos]);
  const handleDelete = (todoID) => {
    const action = {
      type: 'delete',
    };
    action.payload = todos.find((todo) => todo.id === todoID);
    dispatch(action);
  };
  const handleToggle = (todoId) => {
    const action = {
      type: 'update',
    };
    action.payload = todos.find((todo) => todo.id === todoId);
    action.payload.done = !action.payload.done;
    dispatch(action);
  };
  const todosLeft = () => todos.filter((todo) => todo.done !== true).length;
  // const handleAddTodo = (newTodo) => {
  //   const action = {
  //     type: 'update',
  //   };
  //   action.payload = newTodo;
  //   dispatch(action);
  // };

  return (
    <div className="todo">
      <div className="todo-title">
        <span>
          <h1>TODO</h1>
        </span>
        <span>
          <img alt="light-mode" src={img} />
        </span>
      </div>
      <div>
        <div>
          <TodoList
            todos={todos}
            handleToggle={handleToggle}
            handleDelete={handleDelete}
          />
        </div>
        <AddTodo dispatch={dispatch} count={todosLeft()} />
      </div>
    </div>
  );
};
export default TodoApp;
