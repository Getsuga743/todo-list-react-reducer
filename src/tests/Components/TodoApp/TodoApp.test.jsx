/* eslint-disable no-shadow */
/* eslint-disable no-undef */
import React from 'react';
import { shallow, mount } from 'enzyme';
import { act } from '@testing-library/react';
import TodoApp from '../../../Components/TodoApp/TodoApp';
import demoTodos from '../../../fixtures/demoTodos';

describe('pruebas en TodoApp', () => {
  const wrapper = shallow(<TodoApp />);
  Storage.prototype.setItem = jest.fn();
  test('debe de mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('debe de agregar un Todo', () => {
    const wrapper = mount(<TodoApp />);
    act(() => {
      wrapper.find('AddTodo').prop('dispatch')(demoTodos[0]);
    });
    expect(localStorage.setItem).toHaveBeenCalledTimes(1);
  });
});
