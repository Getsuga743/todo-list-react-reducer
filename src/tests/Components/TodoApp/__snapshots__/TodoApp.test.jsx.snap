// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en TodoApp debe de mostrarse correctamente 1`] = `
<div
  className="todo"
>
  <div
    className="todo-title"
  >
    <span>
      <h1>
        TODO
      </h1>
    </span>
    <span>
      <img
        alt="light-mode"
        src="icon-sun.svg"
      />
    </span>
  </div>
  <div>
    <div>
      <TodoList
        handleDelete={[Function]}
        handleToggle={[Function]}
        todos={Array []}
      />
    </div>
    <AddTodo
      count={0}
      dispatch={[Function]}
    />
  </div>
</div>
`;
